frontend/app/components/ChatWindow.tsx

```diff
@@ -38,27 +45,92 @@ const MODEL_TYPES = [
 const defaultLlmValue =
   MODEL_TYPES[Math.floor(Math.random() * MODEL_TYPES.length)];
 
-export function ChatWindow(props: { conversationId: string }) {
-  const conversationId = props.conversationId;
+const getAssistantId = async (client: Client) => {
+  const response = await client.assistants.search({
+    metadata: null,
+    offset: 0,
+    limit: 10,
+  });
+  if (response.length !== 1) {
+    throw Error(
+      `Expected exactly one assistant, got ${response.length} instead`,
+    );
+  }
+  return response[0]["assistant_id"];
+};
 
+export function ChatWindow() {
+  const router = useRouter();
   const searchParams = useSearchParams();
 
   const messageContainerRef = useRef<HTMLDivElement | null>(null);
-  const [messages, setMessages] = useState<Array<Message>>([]);
   const [input, setInput] = useState("");
   const [isLoading, setIsLoading] = useState(false);
   const [llm, setLlm] = useState(
     searchParams.get("llm") ?? "openai_gpt_3_5_turbo",
   );
   const [llmIsLoading, setLlmIsLoading] = useState(true);
+  const [assistantId, setAssistantId] = useState<string>("");
+  const [userId, setUserId] = useLocalStorage("userId", null);
+
+  const client = useLangGraphClient();
+
+  const { currentThread } = useThread(userId);
+  const {
+    threads,
+    createThread,
+    updateThread,
+    deleteThread,
+    loadMoreThreads,
+    areThreadsLoading,
+  } = useThreadList(userId);
+  const { streamState, startStream, stopStream } = useStreamState();
+  const { refreshMessages, messages, setMessages, next, areMessagesLoading } =
+    useThreadMessages(
+      currentThread?.thread_id ?? null,
+      streamState,
+      stopStream,
+    );
+
+  const setLanggraphInfo = async () => {
+    try {
+      const assistantId = await getAssistantId(client);
+      setAssistantId(assistantId);
+    } catch (e) {
+      toast.error("Could not load AI agent");
+    }
+  };
+
+  const setUserInfo = () => {
+    if (userId == null) {
```

Comment:
Any potential race condition with the local storage hook above? I think not but just double checking

Reply:
Could also use some kind of `getOrCreateUserId` pattern and combine it into your other hook?

Reply:
> Any potential race condition with the local storage hook above? I think not but just double checking

don't think so